flink-runtime/src/main/java/org/apache/flink/runtime/rpc/RpcEndpoint.java

```diff
@@ -147,16 +147,19 @@ protected final void stop() {
 	 * @return Future which is completed once all post stop actions are completed. If an error
 	 * occurs this future is completed exceptionally
 	 */
-	public abstract CompletableFuture<Void> postStop();
+	public CompletableFuture<Void> onStop() {
```

Comment:
This would not solve the underlying problem that whenever you have access to the class extending `RpcEndpoint` you would have access to this method as well. In some way, we already have this separation with the `RpcGateway` and `RpcEndpoint`.